packages/compass-user-data/src/user-data.ts

```diff
@@ -252,11 +252,230 @@ export class FileUserData<T extends z.Schema> extends IUserData<T> {
   async updateAttributes(
     id: string,
     data: Partial<z.input<T>>
-  ): Promise<z.output<T>> {
-    await this.write(id, {
-      ...((await this.readOne(id)) ?? {}),
-      ...data,
-    });
-    return await this.readOne(id);
+  ): Promise<boolean> {
+    try {
+      await this.write(id, {
+        ...((await this.readOne(id)) ?? {}),
+        ...data,
+      });
+      return true;
+    } catch {
+      return false;
+    }
+  }
+}
+
+// TODO: update endpoints to reflect the merged api endpoints https://jira.mongodb.org/browse/CLOUDP-329716
+export class AtlasUserData<T extends z.Schema> extends IUserData<T> {
+  private readonly authenticatedFetch;
+  private readonly getResourceUrl;
+  private orgId: string = '';
+  private projectId: string = '';
+  constructor(
+    validator: T,
+    dataType: string,
+    orgId: string,
+    projectId: string,
+    getResourceUrl: GetResourceUrl,
+    authenticatedFetch: AuthenticatedFetch,
+    { serialize, deserialize }: AtlasUserDataOptions<z.input<T>>
+  ) {
+    super(validator, dataType, { serialize, deserialize });
+    this.authenticatedFetch = authenticatedFetch;
+    this.getResourceUrl = getResourceUrl;
+    this.orgId = orgId;
+    this.projectId = projectId;
+  }
+
+  async write(id: string, content: z.input<T>): Promise<boolean> {
+    try {
+      this.validator.parse(content);
+
+      const response = await this.authenticatedFetch(
+        await this.getResourceUrl(
+          `${this.dataType}/${this.orgId}/${this.projectId}`
+        ),
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            id: id,
+            data: this.serialize(content),
+            createdAt: new Date(),
+            projectId: this.projectId,
+          }),
+        }
+      );
+
+      if (!response.ok) {
+        throw new Error(
+          `Failed to post data: ${response.status} ${response.statusText}`
+        );
+      }
```

Comment:
This is already handled by `authenticatedFetch` method, so no need to check ok here